lib/src/git_subprocess.rs

```diff
@@ -444,6 +468,22 @@ fn parse_git_remote_show_output(output: Output) -> Result<Output, GitSubprocessE
     Err(external_git_error(&output.stderr))
 }
 
+fn parse_git_list_remote_output(output: Output) -> Result<Output, GitSubprocessError> {
```

Comment:
Nit: I would call this validate rather than parse. parse implies to me that stdout has been parsed into something useful.

That or make this function return `Result<Vec<(CommitId, GitRefNameBuf>, GitSubprocessError>`, which I'd honestly be happier with anyway.

Reply:
I will probably use the other method `parse_ref_names` in this function to return the result directly.